### PR TITLE
Switch to use android library plugin

### PR DESCRIPTION
**Pull request by tzuhanwu**
This branch is to migrate the SDK to use android-library-plugin to decouple with non-Android Java. Since it's not backward compatible, it will be a major version bump to v3.0.0.

* gradlew files are added so Gradle wrapper is used for compile.
* Minor fix for BoundingBoxOptions.java to add default Locale.
* Add changelog file.
* Bump auto-value version to 1.2.
* Bump jackson-dataformat-yaml version to 2.7.2.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,21 @@
 [![Build Status](https://travis-ci.org/Yelp/yelp-android.svg?branch=master)](https://travis-ci.org/Yelp/yelp-android)
 
 # yelp-android
-A Java library for the Yelp API. It simplifies the process of authentication, request construction, and response 
-parsing for Java/Android developers using the Yelp API. This clientlib requires Java 7 or Android 2.3.
+An Android library for the [Yelp API v2](https://www.yelp.com/developers/documentation/v2/overview). It simplifies the 
+process of authentication, request construction, and response parsing for Android developers using the 
+[Yelp API v2](https://www.yelp.com/developers/documentation/v2/overview). This clientlib has been tested with 
+applications wrote in Android API level 15 and 25.
```
Comment (jbrick) [inline README.md]: applications written

File: gradle/lint.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<lint>
+    <issue id="InvalidPackage">
+    <!-- Some libraries include non-android compatible functions. Lint complains it even though it's not being called
```
Comment (jbrick) [inline gradle/lint.xml]: *Lint complains even though

**Review by jbrick — COMMENTED** [review]
Mostly looks good, just a couple grammar comments.

File: README.md (hunk 1 above)
Comment (tzuhanwu) [inline README.md]: Fixed.

File: gradle/lint.xml (hunk 2 above)
Comment (tzuhanwu) [inline gradle/lint.xml]: Fixed.

**Comment by jbrick** [comment]
ShipIt

**Pull request merged**